packages/node/src/retry/baseRetry.ts

```diff
@@ -0,0 +1,43 @@
+import { Event, Options, Transport, Payload, PayloadOptions, Status, Response, Retry } from '@amplitude/types';
+import { setupDefaultTransport } from '../transports';
+import { DEFAULT_OPTIONS } from '../constants';
+
+export class BaseRetryHandler implements Retry {
+  protected readonly _apiKey: string;
+  protected readonly _options: Options;
+  protected readonly _transport: Transport;
+
+  public constructor(apiKey: string, options: Partial<Options> = {}) {
+    this._apiKey = apiKey;
+    this._options = { ...DEFAULT_OPTIONS, ...options };
+    this._transport = this._options.transportClass ?? setupDefaultTransport(this._options);
+  }
+
+  /**
+   * @inheritDoc
+   */
+  public async sendEventsWithRetry(events: readonly Event[]): Promise<Response> {
+    let response: Response = { status: Status.Unknown, statusCode: 0 };
+    response = await this._transport.sendPayload(this._getPayload(events));
+    return response;
+  }
+
+  protected _getPayloadOptions(): { options?: PayloadOptions } {
+    if (typeof this._options.minIdLength === 'number') {
```

Comment:
noob typescript question here: if `_options` is typed as an `Options`, why specifically check if `minIdLength` is a number? Or is this a roundabout way of making sure it's not null?

Reply:
`minIdLength` is by default `null` since it's typed as `null | number` - if it's not defined, I think it would be better for the HTTP API to decide what the "correct" min ID length is